pylint/checkers/classes.py

```diff
@@ -1352,9 +1353,20 @@ def _check_protected_attribute_access(self, node):
             if self._is_type_self_call(node.expr):
                 return
 
+            # Check if we are inside class or inner class
+            inside_klass = True
+            curr_klass = klass
+            parents_callee = callee.split(".")
+            parents_callee.reverse()
+            for c in parents_callee:
+                if c != curr_klass.name:
+                    inside_klass = False
+                    break
+                curr_klass = get_outer_class(curr_klass)
```

Comment:
`get_outer_class()` can return None. What should the loop do in this case?
I think that the only time this can happen is if the attribute chain doesn't match the class chain so I think `inside_klass` should get set to `False` in this case as well?

Reply:
I am sorry, could you provide a code sample about when the attribute chain doesn't match the class chain?
Because I think when the attribute chain doesn't match the class chain, the if at line 1362 will be true and `inside_klass` will be set to `False`.

The only time `get_outer_class` will return None is when the callee is not a class or it is the last (root) class. The loop will exit on its own.

Reply:
Regardless, I add an if statement at 20a7e69 to avoid potential bug when `get_outer_class` return `None`.